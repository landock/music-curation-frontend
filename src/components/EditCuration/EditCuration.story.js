import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { linkTo } from '@storybook/addon-links';
import EditCuration from '.';

import { curation } from '../../fixtures/test_data';

storiesOf('EditCuration', module).add('default', () => (
  <EditCuration curation={curation} />
));
