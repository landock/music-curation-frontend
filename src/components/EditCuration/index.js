import React from 'react';
import recycle from 'recycle';
import { Grid, Item } from 'semantic-ui-react';

import Curation from '../Curation';
import CurationScheduler from '../CurationScheduler';

import 'rxjs/add/operator/merge';
import 'rxjs/add/operator/first';

import { actions as curationsActions } from '../../modules/Curations';

const EditCuration = recycle({
  dispatch(sources) {
    let { fetchCuration } = curationsActions;
    return [
      sources
        .select('button')
        .addListener('onClick')
        .mapTo({ type: 'UPDATE_CURATION_SCHEDULE', payload: {} }),
      sources.lifecycle
        .filter(e => {
          return e === 'componentDidMount';
        })
        .merge(sources.props)
        .first()
        .map(props => {
          return fetchCuration(props.match.params.id);
        }),
    ];
  },
  update(sources) {
    return [
      sources.store.reducer((state, store) => {
        state.currentCuration = store.Curations.currentCuration;
        return state;
      }),
    ];
  },
  view(props, state) {
    return (
      <Grid columns={2}>
        <Grid.Row>
          <Grid.Column width={10}>
            <Item.Group>
              <Curation curation={state.currentCuration} />
            </Item.Group>
          </Grid.Column>
          <Grid.Column width={6}>
            <CurationScheduler />
          </Grid.Column>
        </Grid.Row>
      </Grid>
    );
  },
});

export default EditCuration;
