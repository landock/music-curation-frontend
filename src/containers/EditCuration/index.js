import React from 'react';
import recycle from 'recycle';

import { getData } from '../../middleware/api';

import { types as curationsTypes } from '../../modules/CurrentCuration';

import { reducer } from 'recycle';
import { withLatestFrom } from 'rxjs/operators';
import EditCurationView from '../../components/EditCurationView';

const EditCuration = recycle({
  //TODO: Add displayName to hygen generator
  displayName: 'EditCuration',
  dispatch(sources) {
    return [getDataOnMount(sources.lifecycle, sources.props)];
  },
  update(sources) {
    return [
      getCurrentCuration(
        sources.store,
        sources.props,
        getCurrentCurationReducer
      ),
    ];
  },
  view(props, state) {
    return <EditCurationView currentCuration={state.currentCuration} />;
  },
});

function getDataOnMount(lifecycleStream, propsStream) {
  return lifecycleStream
    .filter(e => {
      return e === 'componentDidMount';
    })
    .withLatestFrom(propsStream)
    .map(([e, props]) => {
      const url = `/curations/${props.match.params.id}`;
      const nextActionType = curationsTypes.CURATION_FETCHED;

      return getData(url, nextActionType);
    });
}

function getCurrentCuration(storeStream, propsStream, reducerFunc) {
  const result = storeStream.pipe(
    withLatestFrom(propsStream),
    reducer(reducerFunc)
  );
  return result;
}

function getCurrentCurationReducer(state, [store, props]) {
  if (!store.CurrentCuration.entities) return state;

  const { tags } = store.CurrentCuration.entities;

  state.currentCuration =
    store.CurrentCuration.entities.curations[props.match.params.id];

  state.currentCuration.tags = tags;
  return state;
}

export {
  EditCuration as default,
  getDataOnMount,
  getCurrentCuration,
  getCurrentCurationReducer,
};
