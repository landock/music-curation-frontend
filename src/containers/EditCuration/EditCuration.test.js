import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import mockStore from '../../fixtures/mockStore';
import EditCuration from '.';

it('EditCuration: default', () => {
  const store = mockStore({
    CurrentCuration: {
      entities: {
        tracks: { 1: { trackName: 'skldjf' }, 2: { trackName: 'lksdjfl' } },

        curations: {
          2: {
            id: 2,
            imageUrl: 'http://fpoimg.com/200x200',
            description: 'whatever description here',
            tracks: [1, 2],
            name: 'Sprint 14 remix',
          },
        },
      },
    },
  });
  const component = renderer.create(
    <Provider store={store}>
      <EditCuration match={{ params: { id: 2 } }} />
    </Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
