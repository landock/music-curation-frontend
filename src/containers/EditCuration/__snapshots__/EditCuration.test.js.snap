// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditCuration: default 1`] = `
<div
  className="ui two column grid"
>
  <div
    className="row"
  >
    <div
      className="ten wide column"
    >
      <div
        className="ui items"
      >
        <p>
          No Curations
        </p>
      </div>
    </div>
    <div
      className="six wide column"
    >
      <form
        action={undefined}
        className="ui form"
        onSubmit={[Function]}
      >
        <div
          className="fields"
        >
          <div
            className="field"
          >
            <button
              className="ui red button"
              disabled={undefined}
              onClick={[Function]}
              required={undefined}
              role="button"
              tabIndex={undefined}
              type="submit"
            >
              Save Curations
            </button>
          </div>
          <div
            className="field"
          >
            <div
              aria-busy={undefined}
              aria-disabled={undefined}
              aria-expanded={false}
              aria-multiselectable={undefined}
              className="ui selection dropdown"
              content={undefined}
              id="curation-status"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              required={undefined}
              role="listbox"
              tabIndex={0}
              type={undefined}
            >
              <div
                aria-live="polite"
                className="default text"
                role="alert"
              >
                DRAFT
              </div>
              <i
                aria-hidden="true"
                className="dropdown icon"
                onClick={[Function]}
              />
              <div
                className="menu transition"
              >
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={true}
                  className="selected item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    Draft
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    Published
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <p>
          <strong>
            Schedule
          </strong>
        </p>
        <div
          className="fields"
        >
          <div
            className="eight wide field"
          >
            <div
              className="ui left icon input"
              content={undefined}
              defautltValue={undefined}
            >
              <input
                disabled={undefined}
                id="date-start"
                onChange={[Function]}
                required={undefined}
                tabIndex={undefined}
                type="date"
              />
              <i
                aria-hidden="true"
                className="calendar outline icon"
              />
            </div>
          </div>
          <div
            className="eight wide field"
          >
            <div
              className="ui left icon input"
              content={undefined}
            >
              <input
                defaultValue={undefined}
                disabled={undefined}
                id="date-end"
                onChange={[Function]}
                required={undefined}
                tabIndex={undefined}
                type="date"
              />
              <i
                aria-hidden="true"
                className="calendar outline icon"
              />
            </div>
          </div>
        </div>
        <div
          className="field"
        >
          <label
            htmlFor="recommended-time"
          >
            Recommended Time Slot
          </label>
          <div
            className="ui input"
            content={undefined}
          >
            <input
              disabled={undefined}
              id="recommended-time"
              max="24"
              min="0"
              onChange={[Function]}
              required={undefined}
              step="1"
              tabIndex={undefined}
              type="range"
              value={undefined}
            />
          </div>
        </div>
        <div
          className="field"
        >
          <label
            htmlFor={undefined}
          >
            Environment
          </label>
          <div
            aria-busy={undefined}
            aria-disabled={undefined}
            aria-expanded={false}
            aria-multiselectable={undefined}
            className="ui selection dropdown"
            content={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            required={undefined}
            role="listbox"
            tabIndex={0}
            type={undefined}
          >
            <div
              aria-live="polite"
              className="default text"
              role="alert"
            >
              Copy to
            </div>
            <i
              aria-hidden="true"
              className="dropdown icon"
              onClick={[Function]}
            />
            <div
              className="menu transition"
            >
              <div
                aria-checked={false}
                aria-disabled={undefined}
                aria-selected={true}
                className="selected item"
                onClick={[Function]}
                role="option"
                style={
                  Object {
                    "pointerEvents": "all",
                  }
                }
              >
                <span
                  className="text"
                >
                  Production
                </span>
              </div>
              <div
                aria-checked={false}
                aria-disabled={undefined}
                aria-selected={false}
                className="item"
                onClick={[Function]}
                role="option"
                style={
                  Object {
                    "pointerEvents": "all",
                  }
                }
              >
                <span
                  className="text"
                >
                  Development
                </span>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
